functions/src/utils/parsers.ts

```diff
@@ -0,0 +1,46 @@
+import { format } from "date-fns";
+
+export const EMPTY_VALUE = "-";
+
+interface IAudioDescription {
+  beginning: string; // h:mm:ss
+  ending: string; // h:mm:ss
+  type: string;
+  description: string;
+}
+
+/**
+ * Used for Unwanted Parts and Sound Issues to create multi-line comments
+ *
+ */
+export function formatMultilineComment(audioDescriptionList: IAudioDescription[]) {
+  if (!audioDescriptionList || !audioDescriptionList.length) {
+    return "-";
+  }
+  let multiline = "";
+  audioDescriptionList.forEach((elem: IAudioDescription, index: number) => {
+    multiline = multiline
+    + `${elem.beginning}-${elem.ending}:${elem.type} -- ${elem.description}`
+    + ((audioDescriptionList.length === (index + 1)) ? "" : "\n");
+  });
+  return multiline;
+}
+
+export function createUpdateLink(token: string): string {
+  return `http://purebhakti.info/audioseva/form/sound-quality-report?token=${token}`;
+}
+
+export function spreadsheetDateFormat(date: number): string {
+  if (!date) {
+    return "-";
+  }
+  return format(date*1000, "MM/DD/YYYY");
```

Comment:
I'm not sure that this will work in all cases. As each spreadsheet has its own date format. Moreover, dates are stored in google sheets as a float number, please research this and make this code reliable.

Reply:
This works for all inputs that are timestamped which is the case for the sheets I was working on (Allotments and Submissions Database), and the output is "MM/DD/YYYY" string which is what I observed in the Example Spreadsheet here: https://docs.google.com/spreadsheets/d/11dlTfbkuWHlaVQFiZIFr1cdRDilqVB-shJXZWEEHcW0/edit#gid=1103170360

I tested this in Allotments and Submissions and the data string is currently working. The dates look like strings to me. Where are you getting floats from? You mean in the database? Yes, the input is float/number.

I was also thinking that all dates in the database would be timestamp format. This function is only for converting database time stamp to "MM/DD/YYYY" strings for the Spreadsheet. When on the Spreadsheet, this string will shows the calendar when clicked as expected.